tests/integ/conftest.py

```diff
@@ -47,6 +48,89 @@ def print_help() -> None:
     )
 
 
+def set_up_external_access_integration_resources(
+    session, rule1, rule2, key1, key2, integration1, integration2
+):
+    try:
+        # IMPORTANT SETUP NOTES: the test role needs to be granted the creation privilege
+        # log into the admin account and run the following sql to grant the privilege
+        # GRANT CREATE INTEGRATION ON ACCOUNT TO ROLE <test_role>;
+        # prepare external access resource
+        session.sql(
+            f"""
+    CREATE OR REPLACE NETWORK RULE {rule1}
+      MODE = EGRESS
+      TYPE = HOST_PORT
+      VALUE_LIST = ('www.google.com');
+    """
+        ).collect()
+        session.sql(
+            f"""
+    CREATE OR REPLACE NETWORK RULE {rule2}
+      MODE = EGRESS
+      TYPE = HOST_PORT
+      VALUE_LIST = ('www.microsoft.com');
+    """
+        ).collect()
+        session.sql(
+            f"""
+    CREATE OR REPLACE SECRET {key1}
+      TYPE = GENERIC_STRING
+      SECRET_STRING = 'replace-with-your-api-key';
```

Comment:
Are these strings passed in plaintext in the queries?

Reply:
yes, they are passed as plaintext, it's for testing setup so we should be fine.